TELECOM-9570: dialog: get ACK cseq from src leg, rather than dst

benceszigeti opened the pull request.

bcnewlin reviewed (commented):
Is this the fix pulled from upstream?

benceszigeti commented:
@bcnewlin Yes, it’s from b8bbf14 (`upstream/master`). This is the same commit `upstream/3.2` cherry-picked: 00cb359. But now I’ve changed to cherry-pick from 00cb359 (see details below).

(The reason I cherry-picked from `master` is because the `3.2` branch sync was broken due to a code conflict: previously, I accidentally pushed `3.2-genesys` into `3.2` while doing the last `3.2-genesys` general sync with `3.2`. It didn’t cause any issues (basically `3.2-genesys` and `3.2` became equal), so I thought I would investigate and fix it later (that’s why I cherry-picked from `master`), but I just fixed the `3.2` branch now (force pushed `upstream/3.2`). So, in this PR I changed the cherry-pick source from `master` to `upstream/3.2`. Now everything as it should be.

I will rethink our sync process to be able to avoid accidents like this. `3.2` is not a protected branch so the GitHub built-in branch sync works. Probably we just want to make it protected and do the sync with scheduled GitHub actions.)
